package/gluon-mesh-vpn-wireguard/luasrc/lib/gluon/upgrade/405-wireguard

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/lua
+local site = require 'gluon.site'
+local util = require 'gluon.util'
+local uci = require('simple-uci').cursor()
+local iputil = require 'gluon.iputil'
+local sysconfig = require 'gluon.sysconfig'
+
+uci:delete('wireguard', 'mesh_vpn', 'user')
+
+local add_groups
+
+local function generate_section_name(peer)
+  return "mesh_vpn_wg_" .. peer
+end
+
+local function generate_wg_iface_name(peer)
+  return "mesh-vpn-" .. peer
+end
+
+local function add_peer(group, name, config, count)
+  local ip = iputil.mac_to_ip("fe80::/64", sysconfig.primary_mac)
+
+  uci:section('network', 'interface', generate_section_name(name) .. 'm', {
+    type = 'wireguard',
+    ifname = generate_wg_iface_name(count),
+    proto = 'gluon_mesh',
+  })
+
+  uci:section('network', 'interface', generate_section_name(name) .. 's', {
+    ip6addr = ip,
+    ifname = generate_wg_iface_name(count),
+    proto = 'static',
+  })
+
+  uci:section('network', 'interface', generate_section_name(name) .. 'w', {
+    ifname = generate_wg_iface_name(count),
+    proto = 'gluon_wireguard',
+  })
```

Comment:
mainly because a previous version of this patch used these section names as interface names and there size limitations do apply. For uci section names I am not sure if there are any such limits but I do not think there are. Could be changed.